### DISCUSSION TITLE
`-depth` parameter changes colors to black and white and `-scale` makes the file size bigger than it needs to be

### DISCUSSION THREAD

**Discussion by tigerros** [ImageMagick/ImageMagick#6962] [Help] [answered]
### ImageMagick version

7.1.1-23

### Operating system, version and so on

Windows 10

### Description

I ran `magick 16x16.png -alpha off -depth 1 -scale 512 512x512.png`, but there's two problems:

- The resulting image is black and white even though the original image only uses two colors which is enough for a 1-bit depth.
- The file size is 467 bytes compared to 286 bytes using paint.net, which also doesn't change the colors.

What am I missing?

<details>

<summary>Images</summary>

You can download these to verify the file size yourself. GitHub doesn't seem to compress them.

[16x16.png](https://github.com/ImageMagick/ImageMagick/assets/76173380/73f86def-26f1-416d-80d5-35d8358da55b)

ImageMagick's 512x512:

![magick 512x512](https://github.com/ImageMagick/ImageMagick/assets/76173380/c7321c7c-5fc1-4579-ad6b-f40b42865b91)

paint.net's 512x512:

![paint.net 512x512](https://github.com/ImageMagick/ImageMagick/assets/76173380/692df8e2-16b3-4a46-90a2-668f6ab9738e)

</details>

**Comment by fmw42** [comment]
-depth 1 means black and white. colors other than black and white even if only two require more depth

PNG color only supports 8-bit or 16-bit depths.  See https://imagemagick.org/Usage/formats/#png_formats for more details.

Perhaps you want to output to GIF and specify +dither -colors 2

**Reply by tigerros** [reply]
Did you check the image I linked? That's a 1-bit depth image that's not black and white. Check it's properties.

**Reply by fmw42** [reply]
You want to set the bit-depth not the depth.  Try

`convert 291758335-73f86def-26f1-416d-80d5-35d8358da55b.png -define png:color-type=3 -define png:bit-depth=1 x.png`

See the reference I provided above.

**Reply by tigerros** [reply]
~~It's huge now. 6.46kb compared to 286 bytes with paint.net. I'm disappointed because this is a CLI tool without a visual editor, so I was expecting some higher quality.~~ Apparently setting `-quality 1` increases the size of the image... but it's still bigger at 484 bytes.

**Reply by fmw42** [reply]
I do not know how your input PNG was compressed.  Or use optipng or one of the other external tools to compress better. You can also play with the PNG compression. Note it does not follow JPG in range 0-100.  See https://imagemagick.org/script/command-line-options.php#compress and https://imagemagick.org/Usage/formats/#png_non-im

Otherwise, one of the IM developers (or someone more expert than me) will need to comment further.

**Reply by fmw42** [reply]
Oops!  I meant -quality not -compress.  See https://imagemagick.org/script/command-line-options.php#quality

**Reply by tigerros** [reply, marked as answer]
Lowering `-quality` actually hurts the file size when dealing with such small images. The problem was not using `-strip`. Thanks for the bit depth and optipng tips though.

**Reply by fmw42** [reply]
-quality does not follow the JPG style of 0 to 100.   It is two numbers. See the quality documentation PNG is different values than JPG

**Comment by snibgo** [comment]
> What am I missing?

`-depth 1` simplifies the image to one bit per channel. So the only colours that can be represented are black, white, red, blue, green, yellow, magenta and yellow. So your image is converted to the nearest colours, which are black and white. If you don't want to change colours, don't use `-depth`.

The input PNG has a palette, aka colour map, with just two entries. IM does not have a control to specify the size of the output palette.
